File: javascript/leetcode/merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var mergeTwoLists = function(headA, headB) {
    let newRoot = new ListNode()
    let newHead = newRoot
    if (!headA && !headB) {
        return null
    }
    while (headA || headB) {
        if (!headA || (headB && headA.val > headB.val)) {
            newHead.val = headB.val
            headB = headB.next
        } else if (!headB || (headB.val >= headA.val)) {
            newHead.val = headA.val
            headA = headA.next
        }
        if (headA || headB) {
            newHead.next = new ListNode()
            newHead = newHead.next
        }
    }
    return newRoot
};

File: python/leetcode/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, headA, headB):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if headA is None and headB is None:
            return None
        new_root = ListNode(None)
        new_head = new_root
        while headA or headB:
            if headA is None or (headB and headA.val >= headB.val):
                new_head.val = headB.val
                headB = headB.next
            elif headB is None or (headB.val > headA.val):
                new_head.val = headA.val
                headA = headA.next
            if headA or headB:
                new_head.next = ListNode(None)
                new_head = new_head.next
        return new_root
